front/src/App.tsx

```diff
@@ -1,24 +1,97 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, {useEffect, useState} from 'react';
 import './App.css';
+import {
+  DetailedTransaction,
+  HttpTransactionsGateway,
+  Transaction
+} from "./services/HttpTransactionsGateway";
+
+function getPaymentType(paymentTypeId: number) {
+  switch (paymentTypeId) {
+    case 1:
+      return 'CB';
+    case 2:
+      return 'Cash';
+    case 3:
+      return 'Check';
+    default:
+      return 'Unknown';
+  }
+}
 
 function App() {
+  const [transactions, setTransactions] = useState<Array<Transaction> | null>(null);
+  
+  const [filter, setFilter] = useState<string>('');
+  const [filteredTransactions, setFilteredTransactions] = useState<Array<Transaction> | null>(transactions);
+
+  const [selectedTransaction, setSelectedTransaction] = useState<DetailedTransaction | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      const transactions = await HttpTransactionsGateway.getTransactions();
+      setTransactions(transactions);
+      setFilteredTransactions(transactions);
+    })();
+  }, []);
+
+  const handleFilterChange = (e: { target: { value: string; }; }) => {
+    setFilter(e.target.value.toLowerCase());
+    filterTransactions(e.target.value.toLowerCase());
+  };
+
+  const filterTransactions = (filterValue: string) => {
+    if (!transactions){
+      return;
+    }
+
+    const filtered = transactions.filter(
+      (transaction) =>
+        transaction.label.toLowerCase().includes(filterValue)
+    );
+    setFilteredTransactions(filtered);
+  };
+
+  const handleRowClick = async (transaction: Transaction) => {
+    const selectedTransaction = await HttpTransactionsGateway.getTransaction(transaction.id);
+    setSelectedTransaction(selectedTransaction);
+  };
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className="container">
+      <div className="container">
+        <h1>Transaction Table</h1>
+        <input
+          type="text"
+          placeholder="Filter by label"
+          value={filter}
+          onChange={handleFilterChange}
```

Comment:
didn't do the button Submit because it is smoother, but can do easily the button, just need to put this function on the btn instead